### PR TITLE
Upgrade bus and load on connect

### PR DESCRIPTION
**Pull request by SnackSBR**
Anything tagged with [HideFlags.HideAndDontSave](https://docs.unity3d.com/ScriptReference/HideFlags.HideAndDontSave.html) can only be deleted by script.

### CODE REVIEW

File: MoreShipUpgrades/Misc/Tools.cs [hunk 1]
```diff
@@ -189,5 +192,30 @@ public static Color ConvertValueToColor(string hex, Color defaultValue)
                 return defaultValue;
             return color;
         }
+
+        public static void ReplaceSaveWithHost()
+        {
+            if (GameNetworkManager.Instance.isHostingGame) return;
```
Comment (WhiteSpike) [inline MoreShipUpgrades/Misc/Tools.cs:198]: No point checking twice if it's hosting the game if we already did before this call

File: MoreShipUpgrades/Misc/Tools.cs [hunk 2]
```diff
@@ -189,5 +192,30 @@ public static Color ConvertValueToColor(string hex, Color defaultValue)
...
+            var localPlayer = StartOfRound.Instance.localPlayerController;
+            logger.LogDebug("Sharing host savefile on reconnecting...");
+            LguStore.Instance.ShareSaveServerRpc();
+            PlayerControllerB[] players = UnityEngine.Object.FindObjectsOfType<PlayerControllerB>();
+            foreach (var player in players)
+            {
+                if (player.IsHost)
```
Comment (WhiteSpike) [inline MoreShipUpgrades/Misc/Tools.cs:205]: NetworkObjects from host side should always have ``IsHost`` true so I don't understand this comparison. Were you trying to grab the controller assigned to host?

Comment (SnackSBR) [reply MoreShipUpgrades/Misc/Tools.cs:205]: `public static void ReplaceSaveWithHost()
{
    var localPlayer = StartOfRound.Instance.localPlayerController;
    logger.LogDebug("Sharing host savefile on reconnecting...");
    LguStore.Instance.ShareSaveServerRpc();
    PlayerControllerB[] players = UnityEngine.Object.FindObjectsOfType<PlayerControllerB>();
    foreach (var player in players)
    {
        if (!player.IsHost)
        {
            StartOfRound.Instance.StartCoroutine(WaitForSync(localPlayer.playerSteamId));
            break;
        }
    }
}`

that should be it, right? localplayer is the host and get every client to assign the new save.

Comment (SnackSBR) [reply MoreShipUpgrades/Misc/Tools.cs:205]: oh, I see what you talking about. would need to compare networkid probably

**Pull request merged**